### ISSUE TITLE
Custom tiles not found after upgrade.

### ISSUE THREAD

**Issue by mauritsvanrees** [plone/plone.app.mosaic#501]
TLDR: not a bug in Mosaic, but in my custom code. I will create the issue and close it immediately.

After upgrading Mosaic from 2.2.3 to 2.2.5, (Plone 5.2.8 to 5.2.9) my custom tiles are no longer found. The editing UI says "Unknown" instead of for example "Slider". I saw this recently in a Plone 6 upgrade as well I think.
In Plone 5.2 see the Insert menu before:
![Screenshot 2022-08-17 at 12 34 37](https://user-images.githubusercontent.com/210587/185101232-11327d6c-122e-453b-8c89-42ba9f8bbf9c.png)

and after:
![Screenshot 2022-08-17 at 12 32 20](https://user-images.githubusercontent.com/210587/185101265-df588d66-de67-4c10-a63f-48f12b83025c.png)

So the custom Zeelandia ones are gone, and the Properties category is added.
I now notice it is an entire category that is missing, so perhaps individual extra tiles would still be available.

This happens when an upgrade step [applies the entire `registry.xml`](https://github.com/plone/plone.app.mosaic/blob/2.2.5/src/plone/app/mosaic/upgrades.zcml#L433). We do that a lot: in the 2.2.x branch I see ten upgrade steps that apply this file.

It may be better to create upgrade profiles, so we only apply changes. There may be lots of things in the `registry.xml` that may be changed by a Manager, either in the Mosaic configuration panel or in the general configuration registry.

But for this particular case, where does this fail? By removing parts of `registry.xml` I narrowed it down to [these lines](https://github.com/plone/plone.app.mosaic/blob/2.2.5/src/plone/app/mosaic/profiles/default/registry.xml#L685-L726) where the tile categories are set.

Wait a minute. An entire category custom category is missing. Could it be that some custom code overrides a category and this is undone by the upgrade? Sigh. Yes, I have this custom config:

```
  <records interface="plone.app.mosaic.interfaces.IWeightedDict"
           prefix="plone.app.mosaic.tiles_categories.properties">
    <value key="name">zeelandia</value>
    <value key="label">Zeelandia</value>
    <value key="weight">5</value>
  </records>
```

This gives the properties category a different name and a new weight, moving it to the front. And applying the Mosaic registry resets it to the default:

```
  <records interface="plone.app.mosaic.interfaces.IWeightedDict"
           prefix="plone.app.mosaic.tiles_categories.properties">
    <value key="name">properties</value>
    <value key="label">Properties</value>
    <value key="weight">50</value>
  </records>
```

Solution for me: set `prefix="plone.app.mosaic.tiles_categories.zeelandia"` and apply our custom registry settings. Voilá: it is back again.
(Some differences remain then, but that is specific for this project and is something for me to find out.)

I will create this issue so others with the same problem can find it, but then immediately close it.
